torch_geometric/loader/node_loader.py

```diff
@@ -0,0 +1,110 @@
+from typing import Any, Callable, Dict, Iterator, Optional, Tuple, Union
+
+import torch
+
+from torch_geometric.data import Data, HeteroData
+from torch_geometric.data.feature_store import FeatureStore
+from torch_geometric.data.graph_store import GraphStore
+from torch_geometric.loader.base import DataLoaderIterator
+from torch_geometric.loader.utils import (
+    filter_custom_store,
+    filter_data,
+    filter_hetero_data,
+    get_input_nodes,
+)
+from torch_geometric.sampler.base import (
+    BaseSampler,
+    HeteroSamplerOutput,
+    NodeSamplerInput,
+    SamplerOutput,
+)
+from torch_geometric.typing import InputNodes
+
+
+class NodeLoader(torch.utils.data.DataLoader):
+    r"""A data loader that performs neighborhood sampling from node information."""
+    def __init__(
+        self,
+        data: Union[Data, HeteroData, Tuple[FeatureStore, GraphStore]],
+        node_sampler: Optional[BaseSampler] = None,
+        node_sampler_kwargs: Dict[str, Any] = None,
+        initialize_sampler=True,
+        input_nodes: InputNodes = None,
+        transform: Callable = None,
+        filter_per_worker: bool = False,
+        **kwargs,
+    ):
+        # Remove for PyTorch Lightning:
+        if 'dataset' in kwargs:
+            del kwargs['dataset']
+        if 'collate_fn' in kwargs:
+            del kwargs['collate_fn']
+
+        self.data = data
+
+        # Initialize sampler with keyword arguments:
+        # NOTE sampler is an attribute of 'DataLoader':
+        self.node_sampler = node_sampler
+        if initialize_sampler:
```

Comment:
Is there a use case for not doing this?

Reply:
Yeah, see `LightningNodeData` and `LightningLinkData` for examples of this (we initialize a sampler once, and use it for multiple dataloaders).

Reply:
Ahh I see. `Lightning` always gives caes I hadn't thought of. Thanks